Bottom: 9fe3941a961931ba308ff07aff66bbe82620f34e
Top:    6c66df03f21966ba2d7c48e3eb80d4701072a575
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:31:28 +0000

Try to prepend, not append to g_lists. This avoids wasting time traversing lists


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 1be059a..93ac247 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -119,19 +119,19 @@ object3d_set_appearance (object3d *object, appearance *appear)
 void
 object3d_add_edge (object3d *object, edge_ref edge)
 {
-  object->edges = g_list_append (object->edges, (void *)edge);
+  object->edges = g_list_prepend (object->edges, (void *)edge);
 }
 
 void
 object3d_add_vertex (object3d *object, vertex3d *vertex)
 {
-  object->vertices = g_list_append (object->vertices, vertex);
+  object->vertices = g_list_prepend (object->vertices, vertex);
 }
 
 void
 object3d_add_face (object3d *object, face3d *face)
 {
-  object->faces = g_list_append (object->faces, face);
+  object->faces = g_list_prepend (object->faces, face);
 }
 
 
@@ -681,7 +681,7 @@ object3d_from_contours (POLYAREA *contours,
     }
 #endif
 
-    objects = g_list_append (objects, object);
+    objects = g_list_prepend (objects, object);
 
     link = malloc (sizeof (polygon_3d_link));
     pa->user_data = link;
@@ -1351,7 +1351,7 @@ old_object3d_from_board_outline (void)
       splice (cylinder_edges[1], SYM(cylinder_edges[1]));
     }
 
-    board_objects = g_list_append (board_objects, board_object);
+    board_objects = g_list_prepend (board_objects, board_object);
 
   } while (pa = pa->f, pa != board_outline);
